SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialEventTracer.h"
+
+#include <inttypes.h>
+
+#include "Interop/Connection/SpatialTraceEventBuilder.h"
+#include "SpatialGDKSettings.h"
+#include <WorkerSDK/improbable/c_io.h>
+#include <WorkerSDK/improbable/c_trace.h>
+
+DEFINE_LOG_CATEGORY(LogSpatialEventTracer);
+
+using namespace SpatialGDK;
+using namespace worker::c;
+
+void SpatialEventTracer::TraceCallback(void* UserData, const Trace_Item* Item)
```

Comment:
Did we establish it was only possible for this to be run in a single thread? Not just as a current implementation detail.

Reply:
Jon - "It can be called on multiple worker threads, but never concurrently."